astropy/io/fits/column.py

```diff
@@ -1423,7 +1423,10 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
-            fields.append((name, dt))
+            if PY2:  # pragma: py2
+                fields.append((name.encode('utf-8'), dt))
```

Comment:
str/unicode conversion is not smart ;-). The case is maybe rare, I don't know, but your original issue was also not common and not tested. Here I think it will fail if I give something like `Column(name=u'Cœur'.encode('utf8') ...)`